src/ImageSharp/Common/Helpers/Vector512Utilities.cs

```diff
@@ -75,6 +82,34 @@ public static Vector512<byte> Shuffle(Vector512<byte> vector, Vector512<byte> in
         return default;
     }
 
+    /// <summary>
+    /// Performs a conversion from a 512-bit vector of 16 single-precision floating-point values to a 512-bit vector of 16 signed 32-bit integer values.
+    /// Rounding is equivalent to <see cref="MidpointRounding.ToEven"/>.
+    /// </summary>
+    /// <param name="vector">The value to convert.</param>
+    /// <returns>The <see cref="Vector128{Int32}"/>.</returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Vector512<int> ConvertToInt32RoundToEven(Vector512<float> vector)
+    {
+        if (Avx512F.IsSupported)
+        {
+            return Avx512F.ConvertToVector512Int32(vector);
+        }
+
+        if (Avx.IsSupported)
+        {
+            Vector256<int> lower = Avx.ConvertToVector256Int32(vector.GetLower());
+            Vector256<int> upper = Avx.ConvertToVector256Int32(vector.GetUpper());
+            return Vector512.Create(lower, upper);
+        }
+
+        Vector512<float> sign = vector & Vector512.Create(-0.0f);
+        Vector512<float> val_2p23_f32 = sign | Vector512.Create(8388608.0f);
```

Comment:
Can you make `8388608.0f` a named constant and use that in the other VectorXyzUtilities too?
So it's defined three times.

Reply:
I'll have a look. Not sure where to put it.

Reply:
Worth noting that `8388608` is `2^23`, which is the boundary value at which fractional data can no longer be represented (only whole integers at or above this boundary).

Reply:
Never found a place to put this.